solr/core/src/java/org/apache/solr/cli/CLIUtils.java

```diff
@@ -216,12 +217,32 @@ public static String normalizeSolrUrl(CommandLine cli) throws Exception {
   }
 
   /**
-   * Get the ZooKeeper connection string from either the zk-host command-line option or by looking
-   * it up from a running Solr instance based on the solr-url option.
+   * Get the value of the specified CLI option with fallback to system property and default value.
+   *
+   * @param cli the command line
+   * @param option the commons cli {@link Option}
+   * @param sysprop the system property to fall back to
+   * @param defaultValue the default value. Use null if no default value is desired
+   * @return the value of the option or system property or the default value
+   */
+  public static String getCliOptionOrPropValue(
+      CommandLine cli, Option option, String sysprop, String defaultValue) {
+    String value = cli.getOptionValue(option);
+    if (value == null) {
+      value = EnvUtils.getProperty(sysprop, defaultValue);
+    }
+    return value;
+  }
```

Comment:
@epugh In my last commit I generalized the util method to this `getCliOptionOrPropValue` which can be used by any Tool that wishes to read a CLI OPTION with a fallback to env/prop and also a fallback to defaultvalue. Currently I only use it for zkhost. IMO the calling code also becomes more clear, as you will see in the same line both the OPT name and the prop name. Perhaps this could be used also for SOLR_URL?